### DISCUSSION TITLE
Question about rpc user/password

### DISCUSSION THREAD

**Discussion by jjjbrogan** [coinfoundry/miningcore#1614] [Q&A]
On my miningcore setup, I have 4 VM's.  mining-stratum, mining-web, mining-master, and the mining-core which has all the core daemons running.    What are the implications if the rpc user/pass on mining-stratum (in the config.json file) does not match the user/pass on mining-core?, as in this section for bitcoin-cash section of config.json:

"daemons": [
        {
          "host": "daemon",
          "port": 8432,
          "user": "rpc",
          "password": "xxxxxxxx",
          "zmqBlockNotifySocket": "tcp://daemon:38432"
        }
(the daemon reference in /etc/hosts points to the core server)
What would happen if the user/pass did not match what is in the core server?  (for example, i can't stop the daemon without it, I've tried) but was wondering what else could happen if that doesn't match?   Before I try to reset the password I wanted to check.  Thank you.

**Comment by seperatis** [comment]
@jjjbrogan I'm not 100% sure what you are asking.....

Are you asking if you can have different RPC user/pass on different instances of the coin daemon?

**Reply by jjjbrogan** [reply]
I guess what I"m asking is..... what can go wrong if the user/pass for the bitcoin cash in the config.json file on the stratum server does not match the bitcoin cash user/pass on the core server?   When a block is found, does that cause any problems?

**Comment by seperatis** [comment]
@jjjbrogan No, just make sure you have payments Disabled on all other instances -- This is typically how people setup Miningcore clusters.

For your stratum instance, which has it's own Core running, the MC config should be pointed to **that** core instance, _presumably with Payments and API disabled_. 

Therefore, _that instance_ will be used for block submissions only, but **not** payments. The MC instance running on stratum VM will probe the daemon running on mining-stratum VM, and _that_ password needs to match the password you put in that MC config.

The only way this could fail, is if you did something like instead of using localhost daemon on mining-stratum, you put the IP address of mining-master or mining-core as the `"host:"`.....if that case, if you set any other password in MC config other than the exact same RPC password that is set on the mining-core VM, then blocks submissions will fail and payments will fail, and honestly I don't thinking MC would even startup on that instance b/c it wouldn't be able to communicate with the daemon.

TL;DR - yes, you can set different RPC passwords on different instances of Miningcore and daemons, but the daemon RPC password attached to MC instance "X" will need to have the same password input into MC instance "X" config.

Hope that makes sense.

PS - RPC passwords are set locally, and VM / instance "A" has no idea what password for instance "B" is. Just make sure that instance "A" RPC password is input into instance "A" MC config, and so on....

**Reply by jjjbrogan** [reply]
Thanks.  That is helpful.  Yes the only daemon running is the mining-core server (bitcoin, bitcoin cash, and dogecoin daemons running only there and nowhere else.)  so the stratum server is not running any daemon - it all ties to the core server.  I thought it was an error to not have matching user/pass in the stratum that points to the core server since the only place the cores were running was on the core server.    Thanks again

**Reply by seperatis** [reply]
Okay, I think that'll be a problem then. You have mining-stratum VM config.json daemon section "host:" pointed to the IP address of mining-core VM? and are not using the same password for both config.jsons?

**Reply by jjjbrogan** [reply]
I resolved it.  It was my error when trying to do a bitcoin-cli command and not specifying the rpc port for bitcoin-cash.  It was defaulting to bitcoin on the stock port which did not match the password.
